remote-messaging/remote-messaging-impl/src/main/java/com/duckduckgo/remote/messaging/impl/mappers/JsonRemoteMessageMapper.kt

```diff
@@ -109,17 +112,29 @@ class JsonRemoteMessageMapper {
 
     private fun JsonRemoteMessage.map(): RemoteMessage? {
         return runCatching {
-            RemoteMessage(
+            val remoteMessage = RemoteMessage(
                 id = this.id.failIfEmpty(),
                 content = this.content!!.mapToContent(this.content.messageType),
                 matchingRules = this.matchingRules.orEmpty(),
                 exclusionRules = this.exclusionRules.orEmpty()
             )
+            remoteMessage.localizeMessage(this.translations)
         }.onFailure {
             Timber.e("RMF: error $it")
         }.getOrNull()
     }
 
+    private fun RemoteMessage.localizeMessage(translations: Map<String, JsonContentTranslations>?): RemoteMessage {
+        if (translations == null) return this
+
+        val locale = deviceProperties.deviceLocale()
+        return translations[locale.asJsonFormat()]?.let {
+            this.copy(content = this.content.localize(it))
+        } ?: translations[locale.language]?.let {
+            this.copy(content = this.content.localize(it))
+        } ?: this
```

Comment:
nit: wow, it is hard to know what will be returned here :D

Reply:
I like it 💁‍♂️  but managed to simplify a bit.